packages/shared/src/hooks/useCompanionTrigger.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useCallback, useMemo } from 'react';
+import { useGrowthBook } from '@growthbook/growthbook-react';
+import { FeedPostClick } from './feed/useFeedOnPostClick';
+import { Post } from '../graphql/posts';
+import { Features } from '../lib/featureManagement';
+import { useAnalyticsContext } from '../contexts/AnalyticsContext';
+
+import { useExtensionPermission } from './useExtensionPermission';
+import { useLazyModal } from './useLazyModal';
+import { LazyModal } from '../components/modals/common/types';
+import { useActions } from './useActions';
+import { ActionType } from '../graphql/actions';
+import { AnalyticsEvent, Origin } from '../lib/analytics';
+
+type CompanionTriggerProps = {
+  post: Post;
+  index?: number;
+  row?: number;
+  column?: number;
+};
+
+type OpenArticle = (data?: CompanionTriggerProps) => () => Promise<void>;
+
+export type CompanionTrigger = {
+  onFeedArticleClick: FeedPostClick;
+  onPostArticleClick: PostClick;
+};
+
+type PostClick = (e: React.MouseEvent, data: { post: Post }) => Promise<void>;
+
+export default function useCompanionTrigger(
+  customPostClickHandler: (
+    post: Post,
+    index?: number,
+    row?: number,
+    column?: number,
+  ) => Promise<void>,
+): CompanionTrigger {
+  const isExtension = process.env.TARGET_BROWSER;
+  const { trackEvent } = useAnalyticsContext();
+  const { closeModal: closeLazyModal, openModal: openLazyModal } =
+    useLazyModal();
+  const { isActionsFetched, checkHasCompleted, completeAction } = useActions();
+  const gb = useGrowthBook();
+
+  const featureEnabled = useMemo(() => {
+    // gb.isOn triggers enrollment in the experiment
+    // to avoid noise in the data, only call this if we are in the extension
+    if (isExtension)
+      return gb.isOn(Features.EngagementLoopJuly2023Companion.id);
```

Comment:
I find these one line `if` without brackets error prone, especially when function has more then one line because condition can easily be missed if new lines are added. cc @dailydotdev/web-team WDYT? We can maybe even add a lint rule for this so its auto formatted.

Reply:
This one https://eslint.org/docs/latest/rules/curly

Reply:
Yeah agreed, it actually has happened before where someone would accidentally add another element and it breaks the statement.
Good point on the rule :) (I'll add a ticket)